Instagram & TikTok Likes Data Report
Gio Fiore
9/13

1. I am the source of my personal Instagram data because I liked all the posts that will show up on my data.
2. This data was created by me liking posts, this data serves a purpose to Instagram and their advertisers so that they can more accurately provide ads on my page.
3. This data can be reliable because it should be only things that I like.
4. This data may be unreliable because not everything I like on Instagram is accurate to my actual personal likes in real life.

In [16]:
import json
import pandas as pd
import matplotlib as pit

In [14]:
path = "liked_posts.json"

with open(path) as j:
    liked_posts = json.load(j)

In [15]:
liked_posts.keys()

dict_keys(['likes_media_likes'])

In [7]:
likes_df = pd.DataFrame(liked_posts[ 'likes_media_likes'])
likes_df

title                                   string_list_data
0         imfromcleveland  [{'href': 'https://www.instagram.com/p/C_LaT_J...
1          oldrowoutdoors  [{'href': 'https://www.instagram.com/p/C_LfXc9...
2             montreality  [{'href': 'https://www.instagram.com/p/C_Lfq5e...
3            barstoolkent  [{'href': 'https://www.instagram.com/p/C_LfhcI...
4          knockoutalerts  [{'href': 'https://www.instagram.com/p/C_LcZaB...
...                   ...                                                ...
16579           hoodville  [{'href': 'https://www.instagram.com/p/Cx1trF0...
16580                 nfl  [{'href': 'https://www.instagram.com/p/Cx3m01C...
16581         blitzalerts  [{'href': 'https://www.instagram.com/p/Cx3sd2z...
16582  ourgenerationmusic  [{'href': 'https://www.instagram.com/p/Cx3eOJT...
16583                 rap  [{'href': 'https://www.instagram.com/p/Cx2-Xa2...

[16584 rows x 2 columns]

In [39]:
likes_df['string_list_data'].iloc[0][0]

{'href': 'https://www.instagram.com/p/C_LaT_JRHSJ/',
 'value': 'ð\x9f\x91\x8d',
 'timestamp': 1724776161}

In [30]:
likes_df = pd.DataFrame(liked_posts['likes_media_likes'])
likes_df.head()

title                                   string_list_data
0  imfromcleveland  [{'href': 'https://www.instagram.com/p/C_LaT_J...
1   oldrowoutdoors  [{'href': 'https://www.instagram.com/p/C_LfXc9...
2      montreality  [{'href': 'https://www.instagram.com/p/C_Lfq5e...
3     barstoolkent  [{'href': 'https://www.instagram.com/p/C_LfhcI...
4   knockoutalerts  [{'href': 'https://www.instagram.com/p/C_LcZaB...

In [40]:
[x[0]['timestamp'] for x in likes_df['string_list_data']]

[1724776161,
 1724776138,
 1724776134,
 1724776100,
 1724774927,
 1724774917,
 1724772843,
 1724771752,
 1724770328,
 1724770319,
 1724768665,
 1724768659,
 1724768638,
 1724768583,
 1724764578,
 1724764177,
 1724736282,
 1724736275,
 1724736263,
 1724735435,
 1724735432,
 1724735426,
 1724725638,
 1724725635,
 1724725612,
 1724725607,
 1724725599,
 1724725586,
 1724725581,
 1724725571,
 1724722792,
 1724713199,
 1724713002,
 1724712993,
 1724712762,
 1724712400,
 1724709374,
 1724706614,
 1724706585,
 1724706403,
 1724706350,
 1724705713,
 1724705708,
 1724705695,
 1724705670,
 1724705294,
 1724704905,
 1724699576,
 1724698229,
 1724696884,
 1724696858,
 1724696660,
 1724696636,
 1724694655,
 1724693047,
 1724692465,
 1724692443,
 1724691081,
 1724689983,
 1724689584,
 1724689518,
 1724688795,
 1724685274,
 1724685036,
 1724685032,
 1724685016,
 1724684996,
 1724684989,
 1724684984,
 1724684737,
 1724684676,
 1724684656,
 1724684613,
 1724684579,
 1724684564,
 1724684527,
 1724684494,

In [45]:
likes_df['timestamp'] = [x[0]['timestamp'] for x in likes_df['string_list_data']]
likes_df.head()

title                                   string_list_data  \
0  imfromcleveland  [{'href': 'https://www.instagram.com/p/C_LaT_J...   
1   oldrowoutdoors  [{'href': 'https://www.instagram.com/p/C_LfXc9...   
2      montreality  [{'href': 'https://www.instagram.com/p/C_Lfq5e...   
3     barstoolkent  [{'href': 'https://www.instagram.com/p/C_LfhcI...   
4   knockoutalerts  [{'href': 'https://www.instagram.com/p/C_LcZaB...   

    timestamp  
0  1724776161  
1  1724776138  
2  1724776134  
3  1724776100  
4  1724774927

In [87]:
df3.groupby('title').count().sort_values('value', ascending = False).head(10)

href  value  timestamp
title                                    
bleacherreport      867    867        867
hoodville           582    582        582
leaguealerts        572    572        572
barstoolu           563    563        563
overtime            503    503        503
hidden.ny           491    491        491
hyperpopdaily       432    432        432
complex             388    388        388
domislivenews       359    359        359
houseofhighlights   353    353        353

I think that compared to my Instagram likes data, I will have less Facebook like data because I do not use it as much as I use Instagram. I think if I use my TikTok data I will have more data to work with since on TikTok it is almost implied to like videos when you are scrolling through. This would lead to me having more Tiktok like data than Instagram but less than Facebook.

In [10]:
path = "likes_and_reactions_1.json"

with open(path) as j:
    likes_and_reactions_1 = json.load(j)

I chose to use my Facebook data because I could not find my TikTok's like data in the file. This will more easily show how much less I use Facebook than Instagram. 

In [25]:
fblikes_data = pd.DataFrame(likes_and_reactions_1)
fblikes_data.head(10)

timestamp                                               data  \
0  1716641609  [{'reaction': {'reaction': 'LIKE', 'actor': 'G...   
1  1723771592  [{'reaction': {'reaction': 'LIKE', 'actor': 'G...   

                                       title  
0  Giovanni Fiore likes Donna Marie's photo.  
1    Giovanni Fiore likes Dan Fiore's photo.

The data in this DataFrame shows that I do not use Facebook as much as I use Instagram. It meets the tidy principles and can be used to test my hypothesis. It shows that I have only used Facebook to like my parent's posts and I have only ever liked 2 posts on Facebook compared to the 16,584 I have liked on Instagram.

A possible next step for this assignment if I decide to come back would be to find my TikTok like data as I feel that would be a more interesting comparison to my Instagram like data.